ignore/src/hgignore.rs

```diff
@@ -0,0 +1,504 @@
+ *       single  directory are read in alphabetical order, later ones overriding earlier ones. Where mul-
+ *       tiple paths are given below, settings from earlier paths override later ones.
+ *
+ *       On Unix, the following files are consulted:
+ *
+ *       o <repo>/.hg/hgrc (per-repository)
+ *
+ *       o $HOME/.hgrc (per-user)
+ *
+ *       o <install-root>/etc/mercurial/hgrc (per-installation)
+ *
+ *       o <install-root>/etc/mercurial/hgrc.d/\*.rc (per-installation)
+ *
+ *       o /etc/mercurial/hgrc (per-system)
+ *
+ *       o /etc/mercurial/hgrc.d/\*.rc (per-system)
+ *
+ *       o <internal>/default.d/\*.rc (defaults)
+ *
+ *       On Windows, the following files are consulted:
+ *
+ *       o <repo>/.hg/hgrc (per-repository)
+ *
+ *       o %USERPROFILE%\.hgrc (per-user)
+ *
+ *       o %USERPROFILE%\Mercurial.ini (per-user)
+ *
+ *       o %HOME%\.hgrc (per-user)
+ *
+ *       o %HOME%\Mercurial.ini (per-user)
+ *
+ *       o HKEY_LOCAL_MACHINE\SOFTWARE\Mercurial (per-installation)
+ *
+ *       o <install-dir>\hgrc.d\*.rc (per-installation)
+ *
+ *       o <install-dir>\Mercurial.ini (per-installation)
+ *
+ *       o <internal>/default.d/\*.rc (defaults)
+ *      Note   The registry key HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Mercurial is used  when  running
+ *              32-bit Python on 64-bit Windows.
+ *
+ *       On Windows 9x, %HOME% is replaced by %APPDATA%.
+ *
+ *       On Plan9, the following files are consulted:
+ *
+ *       o <repo>/.hg/hgrc (per-repository)
+ *
+ *       o $home/lib/hgrc (per-user)
+ *
+ *       o <install-root>/lib/mercurial/hgrc (per-installation)
+ *
+ *       o <install-root>/lib/mercurial/hgrc.d/\*.rc (per-installation)
+ *
+ *       o /lib/mercurial/hgrc (per-system)
+ *
+ *       o /lib/mercurial/hgrc.d/\*.rc (per-system)
+ *
+ *       o <internal>/default.d/\*.rc (defaults)
+ */
+/// FIXME: parses only ignore setting from $HOME/.hgrc
+fn hgrc_ignore_path() -> Option<PathBuf> {
+    hgrc_contents()
+        .and_then(|data| parse_ignore_file(&data))
+}
+
+fn hgrc_contents() -> Option<Vec<u8>> {
+    let home = match env::var_os("HOME") {
+        None => return None,
+        Some(home) => PathBuf::from(home),
+    };
+    let mut file = match File::open(home.join(".hgrc")) {
+        Err(_) => return None,
+        Ok(file) => io::BufReader::new(file),
+    };
+    let mut contents = vec![];
+    file.read_to_end(&mut contents).ok().map(|_| contents)
+}
+
+
+fn parse_ignore_file(data: &[u8]) -> Option<PathBuf> {
+    lazy_static! {
+        static ref RE: Regex = Regex::new(
+            r"(?ium)^\s*ignore\s*=\s*(.+)\s*$").unwrap();
+    };
+    let caps = match RE.captures(data) {
+        None => return None,
+        Some(caps) => caps,
+    };
+    str::from_utf8(&caps[1]).ok().map(|s| PathBuf::from(expand_tilde(s)))
+}
+
+/// Expands ~ in file paths to the value of $HOME.
+fn expand_tilde(path: &str) -> String {
+    let home = match env::var("HOME") {
```

Comment:
I think you actually want [`env::home_dir`](https://doc.rust-lang.org/std/env/fn.home_dir.html) instead.